bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/extensions/ExtensionResource.java

```diff
@@ -0,0 +1,177 @@
+
+    private final Logger logger = LoggerFactory.getLogger(ExtensionResource.class);
+
+    private ExtensionService extensionService;
+    private EventPublisher eventPublisher;
+
+    protected void setExtensionService(ExtensionService featureService) {
+        this.extensionService = featureService;
+    }
+
+    protected void unsetExtensionService(ExtensionService featureService) {
+        this.extensionService = null;
+    }
+
+    protected void setEventPublisher(EventPublisher eventPublisher) {
+        this.eventPublisher = eventPublisher;
+    }
+
+    protected void unsetEventPublisher(EventPublisher eventPublisher) {
+        this.eventPublisher = null;
+    }
+
+    @Context
+    UriInfo uriInfo;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Get all extensions.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK") })
+    public List<Extension> getExtensions(
+            @HeaderParam("Accept-Language") @ApiParam(value = "language") String language) {
+        logger.debug("Received HTTP GET request at '{}'", uriInfo.getPath());
+        Locale locale = LocaleUtil.getLocale(language);
+        return extensionService.getExtensions(locale);
+    }
+
+    @GET
+    @Path("/types")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Get all extension types.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK") })
+    public List<Type> getTypes(@HeaderParam("Accept-Language") @ApiParam(value = "language") String language) {
+        logger.debug("Received HTTP GET request at '{}'", uriInfo.getPath());
+        Locale locale = LocaleUtil.getLocale(language);
+        return extensionService.getTypes(locale);
+    }
+
+    @GET
+    @Path("/{extensionId: [a-zA-Z_0-9-]*}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Get extension with given ID.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"), @ApiResponse(code = 404, message = "Not found") })
+    public Response getById(@HeaderParam("Accept-Language") @ApiParam(value = "language") String language,
+            @PathParam("extensionId") @ApiParam(value = "extension ID", required = true) String extensionId) {
+        logger.debug("Received HTTP GET request at '{}'.", uriInfo.getPath());
+        Locale locale = LocaleUtil.getLocale(language);
+        Object responseObject = extensionService.getExtension(extensionId, locale);
+        if (responseObject != null) {
+            return Response.ok(responseObject).build();
+        } else {
+            return Response.status(404).build();
+        }
+    }
+
+    @POST
+    @Path("/{extensionId: [a-zA-Z_0-9-]*}/install")
+    @ApiOperation(value = "Installs the extension with the given ID.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 500, message = "Extension cannot be installed due to an internal error") })
```

Comment:
I don´t see in which case a 500 is sent? I guess it is only if the thread can not be started, but you won´t get a 500 if the installation was not successful? Even if the `extensionService.install(extensionId);` directly throws an exception, you will get 200.